src/components/admin/form-components/MaxLenTextField.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { capFirstLetter } from 'lib/utilities';
+import TextField from 'material-ui/TextField';
+
+export default class MaxLenTextField extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onChange = (event) => this.props.onUpdate(event.target.value.substr(0, this.props.maxLen));
+  }
+
+  render() {
+    return (
+      <TextField
+        name={this.props.name}
+        floatingLabelText={
+          `${capFirstLetter(this.props.name)} ` +
+          `(${this.props.value.length} ` +
+          `of ${this.props.maxLen} characters) `
+        }
+        value={this.props.value}
+        disabled={this.props.disabled}
+        onChange={this.onChange}
+        multiLine
+        fullWidth
+      />
+    );
+  }
+}
+
+MaxLenTextField.propTypes = {
+  maxLen: PropTypes.number.isRequired,
+  onUpdate: PropTypes.func.isRequired,
+  value: PropTypes.string,
+  disabled: PropTypes.bool,
+  name: PropTypes.string,
```

Comment:
I guess with Material UI it's okay not providing a name? Or maybe when it's controlled it's also okay right?

Reply:
Yeah its fine without a name.

Reply:
Okay cool, in that case we might not even need this prop right? As I think the name is only needed for uncontrolled components so the form can access them through the onSubmit event